external_parser/joiners/multistep_example_joiner.cc

```diff
@@ -226,8 +220,8 @@ bool multistep_example_joiner::process_joined(v_array<example *> &examples) {
     return false;
   }
 
-  const auto reward = _reward_calculation(joined.outcome_events);
-  try_set_label(joined, reward, examples);
+  joined.calc_reward(_loop_info.default_reward, _reward_calculation.value());
```

Comment:
flipping `fill_in_label` and `calc_reward` like that makes the code much cleaner <3